docs/proposals/high-availability.md

```diff
@@ -0,0 +1,34 @@
+# High Availability of Daemons in Kubernetes
+This document serves as a proposal for high availability of the master daemons in kubernetes.
+
+## Design Options
+1. Hot Standby Daemons: In this scenario, data and state are shared between the two deamons such that an immediate failure in one daemon causes the the standby deamon to take over exactly where the failed daemon had left off.  This would be an ideal solution for kubernetes, however it poses a series of challenges in the case of controllers where daemon-state is cached locally and not persisted in a transactional way to a storage facility.  As a result, we are **NOT** planning on this approach. 
```

Comment:
Agreed, good to get crisp on the definition of cold, warm and hot standby and load-balanced, to avoid confusion.  Something like this:

https://www.ibm.com/developerworks/community/blogs/RohitShetty/entry/high_availability_cold_warm_hot?lang=en